63/Gealber/generate_blog_images.py

```diff
@@ -0,0 +1,128 @@
+from bs4 import BeautifulSoup
+from urllib.request import Request, urlopen, URLError
+from selenium import webdriver
+from selenium.webdriver.support.ui import Select, WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+import random
+import time
+import zipfile
+import os
+import re
+
+
+class Extractor:
+    def __init__(self, headers):
+        self.headers = headers
+
+    @staticmethod
+    def parser(page, query):
+        soup = BeautifulSoup(page, 'lxml')
+        items = soup.select(query)
+        return items
+
+    def extract_challenge(self):
+        url = 'https://codechalleng.es/challenges/'
+        request = Request(url, headers=self.headers)
+        base_url = 'https://codechalleng.es'
+        challenges = {}
+        try:
+            with urlopen(request) as response:
+                html = str(response.read(), 'utf-8')
+            items = self.parser(html, "td.challengeTitle > a")
+            for item in items:
+                link = "".join([base_url, item.get('href')])
+                challenges[item.get_text()] = link
+        except URLError as error:
+            print(error.reason)
+        return challenges
+
+    def extract_articles(self):
+        url = 'https://pybit.es/pages/articles.html'
+        request = Request(url, headers=self.headers)
+        articles = {}
+        try:
+            with urlopen(request) as response:
```

Comment:
`extract_challenge` and `extract_articles` have duplicated code, extract into a helper